100-level-AsyncAPI-Intro/step1.md

```diff
@@ -0,0 +1,54 @@
+Eve assumes her problem-solving role and approaches the spaceship. Inside, Eve discovers Chan, a programming genius visiting from Brownieterra. It becomes apparent that a technical glitch has affected the ship's communication system.
+
+With her expertise in event-driven architecture (EDA) and AsyncAPI, Eve swiftly springs into action. She formulates a plan to establish seamless communication among the spaceship's components in collaboration with Chan.
+
+Eve begins by providing Chan with a comprehensive explanation of event-driven architecture (EDA) and outlines the process of designing an EDA system using AsyncAPI.
+
+## Event-driven architecture
+
+An event signifies a change in the system's state that triggers communication between different services. Event-driven architecture (EDA) decomposes an application into several components interacting via events. These components encompass event producers and consumers:
+
+- An event **producer** identifies an event and transmits relevant information through the event channel.
+- An event **consumer** subscribes to an event channel to receive and process event data.
+
+Each component can fulfill the role of an event producer, consumer, or both.
+
+As Eve continues her conversation with Chan, she realizes she hasn't offered him a beverage. She quickly rushes to prepare a cup of coffee. While making the coffee, she draws upon a coffee shop as an analogy: a customer (producer) places an order for an espresso (event), conveying the order details to the barista (consumer). The sequence unfolds as follows:
+
+1. The customer initiates an espresso order (an event), and the barista receives the order details.
+2. Subsequently, the barista processes the order and begins crafting the espresso (a response to the event).
```

Comment:
I would say there is a missing step here that needs to be added in order to prove it is an asynchronous transaction.
Let me clarify that the customer, if it's waiting in the queue line and didn't pay yet, it is waiting actively that (to pay and get the coffee). I would say this can sound confusing and not be really taken as an asynchronous transaction/process.

What if, after paying, the customer goes to a table and later, once the coffee gets crafted, the barista serves it right directly into the table? Another alternative could be the typical Starbucks model, where you pay and you go to another place to take it once they call you (response) by your name.

Reply:
This example would explain asynchronous transactions better, but maybe it could make the example a bit complicated.

Reply:
Ah, I too struggled at first to incorporate Sergio's feedback on this one. After mulling it over, I also agree with Sergio that a step is missing for it to qualify as asynchronous communication. Here's how I recommend you update it @kakabisht 😸:

Incorporating the technical reviewer's feedback, we can refine the analogy to better illustrate the asynchronous transaction by adding the suggested step. Here's how the revised section could look:

---

As Eve continues her conversation with Chan, she realizes she hasn't yet offered him a beverage. She decides to make a cup of coffee and uses the concept of a coffee shop as an analogy for asynchronous communication:

1. A customer (producer) orders an espresso (event) and pays for it, akin to sending a message in an asynchronous system.
2. After paying, the customer moves to a separate area, such as a table or the waiting zone, freeing up the queue. This separation of order placement and fulfillment zones mirrors the decoupling in asynchronous transactions.
3. The barista (consumer) processes the order and, once the espresso is prepared, delivers it to the customer at their table or calls the customer by name to collect it from the counter. This step signifies the completion of the asynchronous process, where the response (the prepared coffee) is received without the customer actively waiting at the counter.

---

Reply:
Fixed